graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/migration/CassandraMigrator.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.schemafirst.migration;
+
+import com.google.common.collect.ImmutableMap;
+import graphql.GraphqlErrorException;
+import io.stargate.db.datastore.DataStore;
+import io.stargate.db.schema.Schema;
+import io.stargate.db.schema.SchemaEntity;
+import io.stargate.db.schema.Table;
+import io.stargate.db.schema.UserDefinedType;
+import io.stargate.graphql.schema.schemafirst.migration.CassandraSchemaHelper.Difference;
+import io.stargate.graphql.schema.schemafirst.processor.EntityMappingModel;
+import io.stargate.graphql.schema.schemafirst.processor.MappingModel;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+public class CassandraMigrator {
+
+  private final DataStore dataStore;
+  private final MappingModel mappingModel;
+  private final MigrationStrategy strategy;
+
+  public CassandraMigrator(
+      DataStore dataStore, MappingModel mappingModel, MigrationStrategy strategy) {
+    this.dataStore = dataStore;
+    this.mappingModel = mappingModel;
+    this.strategy = strategy;
+  }
+
+  /** @throws GraphqlErrorException if the CQL data model can't be migrated */
+  public List<MigrationQuery> compute() {
+
+    Schema schema = dataStore.schema();
+
+    List<MigrationQuery> queries = new ArrayList<>();
```

Comment:
I've temporarily disabled the sorting because my comparator was broken. I need to revisit this with the topological sort anyway.